### PR TITLE
VB-30: Restrict Modal Closing to Close Button Only

### PR DESCRIPTION
**Pull request by AndrewRiabets**
## Description 
This PR updates the modal interaction logic to prevent accidental data loss. By disabling "click-to-dismiss" on the backdrop, we ensure that users only close modals through explicit actions, while maintaining keyboard accessibility.

## What changed
- **Backdrop Interaction:** Disabled closing modals by clicking outside the modal area.
- **Consistent Navigation:**  Maintained explicit close paths (e.g., X button, Cancel, OK). Unified Esc key behavior across all dialogs, including SweetAlerts.
- **Stacked Modals:** Pressing Esc now correctly closes only the topmost layer (e.g., an Alert on top of a Modal).
- **Z-Index Fixes:** Improved overlay layering to ensure alert dialogs always appear above other elements.
- **Focus Management:** Fixed an edge case where closing a menu via Esc would leave a "stale" focus ring.

Preserved menu dropdown behavior: it still closes when clicking outside of the dropdown area.

## How to Test 
1. Open any modal in the app.
2. Click outside the modal (on backdrop) → modal should remain open.
3. Click the close button (X/Ok/Close etc) → modal should close.
4. Repeat across different modals in the app to ensure consistency.

### CODE REVIEW

**Review by rogergraves — APPROVED** [review]

**Review by PivtoranisV — APPROVED** [review]

File: app/javascript/components/UI/rich-text/RichInputElement.js [hunk 1]
```diff
@@ -83,6 +83,9 @@ const RichInputElement = ({
   }, []);
 
   const handleKeyDown = (event) => {
+    if (event.key === 'Escape' && !isDropdownList) {
```
Comment (vb-cmd) [inline app/javascript/components/UI/rich-text/RichInputElement.js:86]: Small suggestion: consider extracting this into a function and reusing it.

Something like this:

```JSX
const isEscape = (event) => event.key === 'Escape'
```

File: app/javascript/components/UI/modal/Portal.js [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
 const Portal = ({onClose, modalOverlay, elementId = 'overlays' }) => {
+    const portalIdRef = useRef(Symbol('portal-modal'));
 
     const portalElement = document.getElementById(elementId)
 
+    useEffect(() => {
+        portalStack.push(portalIdRef.current);
+
+        const handleKeyDown = (event) => {
+            if (event.key !== 'Escape') return;
```
Comment (vb-cmd) [inline app/javascript/components/UI/modal/Portal.js:16]: Also here

**Review by vb-cmd — APPROVED** [review]
